### PR TITLE
Add Amazon Chime Echo Reduction support in Voice Focus.

### PR DESCRIPTION
**Pull request by dimitarz**
**Issue #:** 

**Description of changes:** 
Adding Echo Reduction improvements to Voice Focus. Echo reduction helps keep echoes—sounds from a user’s loudspeaker that get picked up by their microphone—from circulating back into meeting audio and bringing discussions to a standstill. The echo reduction feature is available in conjunction with the noise reduction provided by Amazon Voice Focus. 

**Testing:**

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*

Added an option to enable this via the Demo app. To enable it, start the demo app and click on 'Enable Echo Reduction (new meetings only)'. This will create a new meeting with the correct attributes such that clients which connect can enable Echo Reduction.

To enable the feature, one must create the meeting as follows:
```typescript
// Create meeting 
const meetingInfo = await chime.createMeeting({
  ...
  MeetingFeatures: {
    Audio: {
      EchoReduction: 'AVAILABLE' 
    }
  } 
}).promise();

// Add attendee 
const attendeeInfo = await chime.createAttendee({...});
const joinInfo = { 
  JoinInfo: {
    Meeting: meetingInfo,
    Attendee: attendeeInfo,
  }
}
```
The client then has to explicitly turn on Echo Reduction as follows:
```typescript
// Select the Echo Reduction model
const spec: VoiceFocusSpec = {
  name: 'ns_es',
  ...
};

// Create the Voice Focus device
const transformer = VoiceFocusDeviceTransformer.create(spec, { logger }, joinInfo);

this.audioVideo = this.meetingSession.audioVideo;
const vfDevice = await transformer.createTransformDevice(chosenAudioInput);

// Enable Echo Reduction on this client
await vfDevice.observeMeetingAudio(this.audioVideo);
```

**Checklist:**

1. Have you successfully run `npm run build:release` locally? Yes.

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved? Yes.

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved? No and no.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. Yes

### CODE REVIEW

File: guides/09_Amazon_Voice_Focus.md [hunk 1]
```diff
@@ -213,13 +260,76 @@ As noted above, the device controller must support Web Audio.
...
 
+To use Amazon Voice Focus, the name section of the specification should be set to `'default'` whereas to use Echo Reduction, the name section of the specification should be set to `'ns_es'`. Echo Reduction can only be enabled while creating the meeting.
+
+```typescript
+spec = { ...
+   name: 'default' // for Amazon Voice Focus
+   ...
+}
+spec = { ...
+   name: 'ns_es' // for Echo Reduction
```
Comment (ltrung) [inline guides/09_Amazon_Voice_Focus.md:271]: ns_es is for both right VF and Echo Reduction?

File: guides/09_Amazon_Voice_Focus.md [hunk 2]
```diff
@@ -213,13 +260,76 @@ As noted above, the device controller must support Web Audio.
...
+
+```typescript
+// Select the Echo Reduction model
+const spec: VoiceFocusSpec = {
+  name: 'ns_es',
+  ...
+};
+
+// Create the Voice Focus device
+const transformer = VoiceFocusDeviceTransformer.create(spec, { logger }, joinInfo);
```
Comment (ltrung) [inline guides/09_Amazon_Voice_Focus.md:321]: Aren't these need to be meetingResp and AttendeeResp and not just joinInfo?

File: guides/09_Amazon_Voice_Focus.md (hunk 1 above)
Comment (dimitarz) [reply guides/09_Amazon_Voice_Focus.md:271]: yes

File: guides/09_Amazon_Voice_Focus.md (hunk 2 above)
Comment (dimitarz) [reply guides/09_Amazon_Voice_Focus.md:321]: VoiceFocusDeviceTransformer can handle both a JoinInfo object or individual ones

File: guides/09_Amazon_Voice_Focus.md (hunk 1 above)
Comment (ltrung) [reply guides/09_Amazon_Voice_Focus.md:271]: can we update it to mention that. It seems we refer to `VF and Echo Reduction` everywhere but not here.

File: demos/serverless/src/handlers.js [hunk 3]
```diff
@@ -17,6 +17,21 @@ const chime = new AWS.Chime({ region: 'us-east-1' });
 // Set the AWS SDK Chime endpoint. The global endpoint is https://service.chime.aws.amazon.com.
 chime.endpoint = new AWS.Endpoint(process.env.CHIME_ENDPOINT);
 
+const chimeRegional = new AWS.ChimeSDKMeetings({ region: 'us-east-1' });
+if (process.env.CHIME_ENDPOINT_REGIONAL) {
```
Comment (ltrung) [inline demos/serverless/src/handlers.js:21]: Where is this defined?

File: demos/browser/server.js [hunk 4]
```diff
@@ -19,25 +19,40 @@ console.info('Using index path', indexPagePath);
...
 // the meeting is hosted in.
 const chime = new AWS.Chime({ region: 'us-east-1' });
-const sts = new AWS.STS({ region: 'us-east-1' })
-
 // Set the AWS SDK Chime endpoint. The global endpoint is https://service.chime.aws.amazon.com.
 const endpoint = process.env.ENDPOINT || 'https://service.chime.aws.amazon.com';
-console.info('Using endpoint', endpoint);
 
+const chimeRegional = new AWS.ChimeSDKMeetings({ region: 'us-east-1' });
+const chimeRegionalEndpoint = process.env.REGIONAL_ENDPOINT || 'https://meetings-chime.us-east-1.amazonaws.com';
```
Comment (ltrung) [inline demos/browser/server.js:30]: I don't think you need to defined the endpoint.

File: demos/browser/server.js [hunk 5]
```diff
@@ -19,25 +19,40 @@ console.info('Using index path', indexPagePath);
...
-// Create ans AWS SDK Chime object. Region 'us-east-1' is currently required.
+// Create ans AWS SDK Chime object. Region 'us-east-1' is globally available..
 // Use the MediaRegion property below in CreateMeeting to select the region
 // the meeting is hosted in.
 const chime = new AWS.Chime({ region: 'us-east-1' });
-const sts = new AWS.STS({ region: 'us-east-1' })
-
 // Set the AWS SDK Chime endpoint. The global endpoint is https://service.chime.aws.amazon.com.
 const endpoint = process.env.ENDPOINT || 'https://service.chime.aws.amazon.com';
-console.info('Using endpoint', endpoint);
```
Comment (ltrung) [inline demos/browser/server.js:30]: This should just be using global endpoint.

**Review by ltrung — DISMISSED** [review]

**Review by ltrung — APPROVED** [review]

**Pull request merged**